test/fluree/db/transact/transact_test.clj

```diff
@@ -213,102 +228,106 @@
-                                                 :bar  "http://bar.com/"
-                                                 :quux "http://quux.com/"}]))))))
+      (let [txn {"@context" ["https://ns.flur.ee"
+                             ""
+                             {:f "https://ns.flur.ee/ledger#"}]
+                 "ledger"   ledger-name
+                 "insert"   [{:context    {:quux "http://quux.com/"}
+                              :id         :ex/alice
+                              :quux/corge "grault"}]}
+            db  @(fluree/transact!2 conn txn)]
+        (is (= #{{:id          :ex/bob
+                  :type        :ex/User
+                  :schema/name "Bob"
+                  :foo/baz     "baz"}
+                 {:id               :ex/alice
+                  :type             :ex/User
+                  :schema/name      "Alice"
+                  :schema/givenName "Alicia"
+                  :quux/corge       "grault"
+                  :foo/bar          "foo"}}
+               (set @(fluree/query db (assoc user-query
+                                             :context ["" {:foo  "http://foo.com/"
+                                                           :bar  "http://bar.com/"
+                                                           :quux "http://quux.com/"}])))))))
     (testing "conn default context is inherited when requested (and not o/w)"
-      (let [txn1 {:context  ["" {:f "https://ns.flur.ee/ledger#"}]
-                  :f/ledger ledger-name
-                  :graph    [{:id      :ex/foo
-                              :type    :ex/Person
-                              :ex/name "Foo"}]}
-            txn2 {:context  {:f "https://ns.flur.ee/ledger#"}
-                  :f/ledger ledger-name
-                  :graph    [{:id      :ex/bar
-                              :type    :ex/Person
-                              :ex/name "Bar"}]}
-            db1  @(fluree/transact! conn txn1 nil)]
+      (let [txn1 {"@context" "https://ns.flur.ee"
+                  "ledger"   ledger-name
+                  "insert"   [{:id      :ex/foo
+                               :type    :ex/Person
+                               :ex/name "Foo"}]}
+            txn2 {"@context" ["https://ns.flur.ee" {}]
+                  "ledger"   ledger-name
+                  "insert"   [{:id      :ex/bar
+                               :type    :ex/Person
+                               :ex/name "Bar"}]}
+            db1  @(fluree/transact!2 conn txn1)]
         (is (= [{:id :ex/foo, :type :ex/Person, :ex/name "Foo"}]
                @(fluree/query db1 '{:select {?p [:*]}
                                     :where  {:id ?p, :type :ex/Person}})))
-        (is (thrown-with-msg? Exception #"Invalid compact-iri"
-                              @(fluree/transact! conn txn2 nil)))))
+        (is (= "Invalid compact-iri: :id Error at idx: [0 :id]"
+               (ex-message @(fluree/transact!2 conn txn2))))))
```

Comment:
Why is this preferable to `thrown-with-msg?`?

Reply:
If I recall it was producing a StackOverflowError every time I ran it when it was erroneously not producing an error, and it would freeze emacs for several seconds every time, so I just changed it to something that would fail fast.